server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java

```diff
@@ -907,10 +925,33 @@ private void publishVMUsageUpdateResourceCount(final UserVm userVm, ServiceOffer
     }
 
     private UserVm importVirtualMachineInternal(final UnmanagedInstanceTO unmanagedInstance, final String instanceName, final DataCenter zone, final Cluster cluster, final HostVO host,
-                                                final VirtualMachineTemplate template, final String displayName, final String hostName, final Account caller, final Account owner, final Long userId,
+                                                final VMTemplateVO template, final String displayName, final String hostName, final Account caller, final Account owner, final Long userId,
```

Comment:
Can we keep `VirtualMachineTemplate` here? `VMTemplateVO` is an implementation of that interface, if the explicit cast is needed it can be done previously checking with a check such as: `if (template instanceof VMTemplateVO) ....`

Reply:
done